test/unit/domain/consents/revoke.test.ts

```diff
@@ -0,0 +1,229 @@
+        .toBe(true)
+    })
+
+    it('Should return false because consent is null', (): void => {
+      expect(isConsentRequestInitiatedByValidSource(
+        null as unknown as Consent, request))
+        .toBeFalsy()
+    })
+
+    it('Should return false because initiator ID does not match', (): void => {
+      expect(
+        isConsentRequestInitiatedByValidSource(partialConsentActive2, request))
+        .toBeFalsy()
+    })
+
+    it('Should throw an error as request headers are missing', (): void => {
+      expect((): void => {
+        isConsentRequestInitiatedByValidSource(
+          partialConsentActive, requestNoHeaders as Request)
+      }).toThrowError()
+    })
+  })
+
+  describe('revokeConsentStatus', (): void => {
+    it('Should return a revoked consent', async (): Promise<void> => {
+      const revokedConsent = await revokeConsentStatus(partialConsentActive)
+      expect(revokedConsent.status).toBe('REVOKED')
+      expect(revokedConsent.revokedAt).toBeDefined()
+      expect(mockConsentUpdate).toHaveBeenCalled()
+      expect(mockLoggerPush).not.toHaveBeenCalled()
+    })
+
+    it('Should also return a revoked consent', async (): Promise<void> => {
+      const revokedConsent = await revokeConsentStatus(completeConsentActive)
+      expect(revokedConsent.status).toBe('REVOKED')
+      expect(revokedConsent.revokedAt).toBeDefined()
+      expect(mockConsentUpdate).toHaveBeenCalled()
+      expect(mockLoggerPush).not.toHaveBeenCalled()
+    })
+
+    it('Should return the consent without any operations, if already revoked',
+      async (): Promise<void> => {
+        const revokedConsent = await revokeConsentStatus(completeConsentRevoked)
+        expect(revokedConsent).toStrictEqual(completeConsentRevoked)
+        expect(mockLoggerPush).toHaveBeenCalled()
+        expect(mockConsentUpdate).not.toHaveBeenCalled()
+      })
+
+    it('Should throw an error due to error in updating consent',
+      async (): Promise<void> => {
+        mockConsentUpdate.mockRejectedValue(new Error('Test Error'))
+
+        await expect(revokeConsentStatus(partialConsentActive2))
+          .rejects
+          .toThrowError('Test Error')
+
+        expect(mockConsentUpdate).toHaveBeenCalled()
+        expect(mockLoggerPush).not.toHaveBeenCalled()
+      })
+  })
+
+  describe('generatePatchConsentRequest', (): void => {
+    it('Should return correct request body', (): void => {
+      expect(generatePatchConsentRequest(completeConsentRevoked))
+        .toStrictEqual(requestBody)
+    })
+
+    it('Should also return correct request body', async (): Promise<void> => {
+      expect(generatePatchConsentRequest(partialConsentRevoked))
+        .toStrictEqual(requestBody)
+    })
+
+    it('Should throw an error as consent is null value', (): void => {
+      expect((): void => {
+        generatePatchConsentRequest(
+          null as unknown as Consent)
+      }).toThrow()
+    })
+  })
```

Comment:
there is a lack of tests where `generatePatchConsentRequest` is called with ACTIVE Consent arg.